ethpm/backends/registry.py

```diff
@@ -97,8 +100,17 @@ def parse_registry_uri(uri: str) -> RegistryURI:
         raise CannotHandleURI(
             f"Invalid address or ENS domain found in uri: {uri}."
         )
-    parsed_name = parsed_uri.path.strip("/")
-    parsed_version = parsed_uri.query.lstrip("version=").strip("/")
-    pkg_name = parsed_name if parsed_name else None
-    pkg_version = parsed_version if parsed_version else None
+    pkg_name, pkg_version = process_pkg_path(parsed_uri.path)
     return RegistryURI(address, chain_id, pkg_name, pkg_version, ens)
+
+
+def process_pkg_path(pkg_path: str) -> Tuple[str, str]:
+    pkg_id = pkg_path.strip("/")
+    if not pkg_id:
+        return None, None
+
+    if "@" not in pkg_id:
+        return pkg_id, None
+    pkg_name, safe_pkg_version = pkg_id.split("@")
```

Comment:
This line would still be executed if `pkg_id = `a@b@c`.

Since `version` is technically allowed to be any string (i think) then I think you need to do this with `pkg_id.partition('@')` which *might* result in a `safe_pkg_version` with multiple `@` symbols in it.

Reply:
This case is covered in `validate_registry_uri` - which will raise an exception if there is more than 1 `@` symbol - since at this stage all `@` symbols should be properly escaped (i.e. `"safe"`). There's test cases covering both ...
- invalid paths: `pkg@1@0` (version= `1@0`)
- valid paths: `pkg@1%400` (version= `1@0`)